src/main/java/org/zeromq/ZMQ.java

```diff
@@ -3173,6 +3174,27 @@ public int sendByteBuffer(ByteBuffer data, int flags)
             return Stream.generate(this::recv);
         }
 
+        /**
+         * This API is in DRAFT state and is subject to change at ANY time until declared stable
+         * handle incoming message with a handler
+         *
+         * @param handler handler to handle incoming message
+         * @param flags   either:
+         *                <ul>
+         *                <li>{@link org.zeromq.ZMQ#DONTWAIT DONTWAIT}:
+         *                Specifies that the operation should be performed in non-blocking mode.
+         *                If there are no messages available on the specified socket,
+         *                the method shall fail with errno set to EAGAIN and return null.</li>
+         *                <li>0 : receive operation blocks until one message is successfully retrieved,
+         *                or stops when timeout set by {@link #setReceiveTimeOut(int)} expires.</li>
+         *                </ul>
+         */
+        @Draft
+        public void recv(Consumer<ZMsg> handler, int flags)
```

Comment:
Have you considered using a promise-like API, as CompletableFuture, or an additional callback for handling exceptions?

Reply:
I have added a new method with an exceptionHandler callback, Thanks for your point